@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -0,0 +1,170 @@
+        const viewDistance0 = refPosition.distanceTo(tempPoiPosition);
+
+        tempPoiPosition
+            .copy(textElement.points[textElement.points.length - 1])
+            .add(textElement.tileCenter!);
+        const viewDistance1 = refPosition.distanceTo(tempPoiPosition);
+
+        viewDistance = Math.min(viewDistance0, viewDistance1);
+    } else {
+        tempPoiPosition.copy(textElement.position).add(textElement.tileCenter!);
+        viewDistance = refPosition.distanceTo(tempPoiPosition);
+    }
+
+    return viewDistance;
+}
+
+/**
+ * Computes the maximum view distance for text elements as a ratio of the given view's maximum far
+ * plane distance.
+ * @param mapView The view for which the maximum view distance will be calculated.
+ * @param farDistanceLimitRatio The ratio to apply to the maximum far plane distance.
+ * @returns Maximum view distance.
+ */
+export function getMaxViewDistance(mapView: MapView, farDistanceLimitRatio: number): number {
+    return mapView.viewRanges.maximum * farDistanceLimitRatio;
+}
+
+/**
+ * State of fading.
+ */
+export enum PrePlacementResult {
+    Ok = 0,
+    NotReady,
+    Invisible,
+    TooFar,
+    Duplicate,
+    Count
+}
+
+/**
+ * Applies early rejection tests for a given text element meant to avoid trying to place labels
+ * that are not visible, not ready, duplicates etc...
+ * @param textElement The Text element to check.
+ * @param tile The tile to which the text element belongs.
+ * @param worldOffsetX The tile's X offset.
+ * @param mapView The view where the text element is meant to be placed.
+ * @param textElementCache The text element cache, needed to deduplicate text elements.
+ * @param [maxViewDistance] If specified, text elements farther than this max distance will be
+ * rejected.
+ * @param [lastFrameNumber] Last frame number when the text element was placed (if any).
+ * @param [stats] If specified, it'll be used to accumulate statistics about the applied tests.
+ * @returns A tuple with the result code as first element, and the text element view distance
+ * ( or `undefined` of the checks failed) as second.
+ */
+export function checkReadyForPlacement(
+    textElement: TextElement,
+    tile: Tile,
+    worldOffsetX: number,
+    mapView: MapView,
+    textElementCache: TextElementStateCache,
+    maxViewDistance?: number,
+    lastFrameNumber?: number
+): [PrePlacementResult, number | undefined] {
+    if (!textElement.visible) {
+        return [PrePlacementResult.Invisible, undefined];
+    }
+
+    // If a PoiTable is specified in the technique, the table is required to be
+    // loaded before the POI can be rendered.
+    if (!mapView.poiManager.updatePoiFromPoiTable(textElement)) {
+        // PoiTable has not been loaded, but is required to determine
+        // visibility.
+        return [PrePlacementResult.NotReady, undefined];
+    }
+
+    if (
+        !textElement.visible ||
```

Comment:
I thought the same, but then I looked into updatePoiFromPoiTable() called above and I noticed it might change the visible flag and the min,max zoom levels of the label.

Reply:
done